### PR TITLE
Multi-determinant orbital rotation.

### PR DESCRIPTION
**Pull request by sapatha2**

### CODE REVIEW

File: src/method/Test_method.cpp [hunk 1]
```diff
@@ -174,7 +174,10 @@ void Test_method::run(Program_options & options, ostream & output)
   }
   
   cout << "#############checking derivatives" << endl;
-  
+  /*
```
Comment (lkwagner) [inline src/method/Test_method.cpp]: This comment isn't relevant to Test_method. Please remove.

File: SUGGESTIONS [hunk 2]
```diff
@@ -0,0 +1,3 @@
+1. Minimize energy of HF for TiO with orbital optimization
```
Comment (lkwagner) [inline SUGGESTIONS]: Please remove SUGGESTIONS

File: tests/c/c.opt_pfaff [hunk 3]
```diff
@@ -0,0 +1,19 @@
+include c.sys
+trialfunc { slater-jastrow
+wf1 { include c.pfaff } 
+wf2 { include c.jast2 } 
+ }
+
+method { VMC nconfig 2000 nstep 10 nblock 10 timestep 1.0 storeconfig ch4.config } 
+
+method { OPTIMIZE2 nconfig 2000 readconfig ch4.config minfunction MIXED}
```
Comment (lkwagner) [inline tests/c/c.opt_pfaff]: Why are we using OPTIMIZE2 here?? This is very outdated.

File: src/method/Test_method.cpp (hunk 1 above)
Comment (sapatha2) [inline src/method/Test_method.cpp]: Done.

File: SUGGESTIONS (hunk 2 above)
Comment (sapatha2) [inline SUGGESTIONS]: Done.

File: tests/c/c.opt_pfaff (hunk 3 above)
Comment (sapatha2) [inline tests/c/c.opt_pfaff]: Not sure, I never used this file for anything.

File: src/method/Linear_optimization.cpp [hunk 4]
```diff
@@ -92,22 +92,33 @@ void Linear_optimization_method::run(Program_options & options, ostream & output
   if(!wfdata->supports(parameter_derivatives))
     error("Wavefunction needs to supports analytic parameter derivatives");
   
-  
+  cout<<"nparms="<<nparms<<endl;
```
Comment (lkwagner) [inline src/method/Linear_optimization.cpp]: No bare cout's please.

File: src/orbitals/MO_matrix_basfunc.cpp.orig [hunk 5]
```diff
@@ -0,0 +1,380 @@
+/*
```
Comment (lkwagner) [inline src/orbitals/MO_matrix_basfunc.cpp.orig]: Please get rid of this file...

File: utils/autogen/cif2crystal.py.orig [hunk 6]
```diff
@@ -0,0 +1,445 @@
+from __future__ import print_function
```
Comment (lkwagner) [inline utils/autogen/cif2crystal.py.orig]: Remove this file.

File: tests/n2/qw.slater [hunk 7]
```diff
@@ -1,16 +1,42 @@
-  SLATER 
-  ORBITALS {
-  CUTOFF_MO
-    MAGNIFY 1
-    NMO 5
-    ORBFILE qw.orb
-    INCLUDE qw.basis
-    CENTERS { USEGLOBAL } 
-  }
+SLATER
```
Comment (lkwagner) [inline tests/n2/qw.slater]: Doesn't this break the N2 test?

**Pull request merged**